### DISCUSSION TITLE
bug: fprettify does nothing ?

### DISCUSSION THREAD

**Discussion by pums974** [krvajal/vscode-fortran-support#669] [Q&A - deprecated see Fortran Discourse] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have defined my formatter to use fprettify but when I "Format Document" nothing happens.

If I do `fprettify <path>` in the terminal, my file get correctly formatted.

I haven't found any message in the output of the extension so I have no idea what is happening

### Expected Behaviour

when I "Format Document", my document should be correctly formatted

### Version of Modern Fortran

3.2.0

### Version of Visual Studio Code

1.71.0

### Platform and Architecture

ArchLinux

**Comment by gnikit** [comment]
I don't think that's a bug, formatters are unittested on our CI. It's more likely that the settings are not being passed correctly. What settings are you using? Please check the README on how to setup formatters.

**Comment by pums974** [comment]
In the settings.json file of the workspace I have
```json
...
    "fortran.linter.compiler": "Disabled",
    "fortran.formatting.formatter": "fprettify",
    "fortran.formatting.path": "venv/bin/fprettify",
    "fortran.formatting.fprettifyArgs": [
        "-c .fprettify.rc"
    ],
    "fortran.fortls.maxLineLength": 120,
...
```

I tried without the fprettifyArgs, and various path for fprettify (absolute, with or without the executable name)

But I haven't found any output what so ever in order to understand what is happening.

**Comment by gnikit** [comment, marked as answer]
> ```json
>     "fortran.formatting.path": "venv/bin/fprettify",
> ```

This is the problem, something like this should work

```json
    "fortran.formatting.path": "./venv/bin",
```

**Comment by pums974** [comment]
Thanks, you are right.

But I had another problem, I must also have

```json
    "fortran.formatting.fprettifyArgs": [
        "-c",
        ".fprettify.rc"
    ],
```

Now it is working correctly. Thanks
